### PR TITLE
[RHCLOUD-24838] feature tests splunk integration

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
Adds tests for the Splunk Integration.

### CODE REVIEW

File: splunk-quarkus/src/test/resources/application.properties [hunk 1]
```diff
@@ -1,3 +1,12 @@
 # CloudWatch needs to be disabled since otherwise Quarkus will ask for the AWS
 # key on "@QuarkusTest"s.
 quarkus.log.cloudwatch.enabled=false
+
+# Set up the Kafka dev service to avoid the thousand-and-one messages about
+# Apache camel not being able to connect to Kafka.
+quarkus.kafka.devservices.enabled = true
+quarkus.kafka.devservices.port = 9092
```
[the remark this answers is not in the document]
Comment (MikelAlejoBR) [reply splunk-quarkus/src/test/resources/application.properties]: Done! Thanks!

File: splunk-quarkus/src/test/resources/application.properties [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
+
+# Disable Micrometer in the tests because we are not testing anything related
+# to it.
```
[the remark this answers is not in the document]
Comment (MikelAlejoBR) [reply splunk-quarkus/src/test/resources/application.properties:11]: Do I leave it, then?

File: splunk-quarkus/pom.xml [hunk 3]
```diff
@@ -164,6 +164,12 @@
             <artifactId>insights-notification-schemas-java</artifactId>
             <version>0.20</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-integration-wiremock-support</artifactId>
+            <version>2.16.0</version>
+            <scope>test</scope>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (MikelAlejoBR) [reply splunk-quarkus/pom.xml:173]: Aaah... Maybe that might be causing the issues with the tests too... I tried swapping the deps but you're right, it kind of broke them.

File: splunk-quarkus/src/test/java/com/redhat/console/integrations/splunk/SplunkIntegrationInvalidUrlTest.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        // Create the payload.
+        final JsonObject cloudEvent = CloudEventTestHelper.buildTestCloudEvent();
+        final JsonObject data = cloudEvent.getJsonObject(CloudEventTestHelper.FIELD_DATA);
+
+        // Update the "url" field and set an invalid URL, which should be
+        // rejected by the "TargetUrlValidator".
+        JsonObject notifMetadata = data.getJsonObject(CloudEventTestHelper.FIELD_NOTIF_METADATA);
+        notifMetadata.put(CloudEventTestHelper.FIELD_NOTIF_METADATA_URL, "https://Hello, World!");
+
+        final Exchange invalidSchemeExchange = createExchangeWithBody(cloudEvent.toString());
```
[the remark this answers is not in the document]
Comment (MikelAlejoBR) [reply splunk-quarkus/src/test/java/com/redhat/console/integrations/splunk/SplunkIntegrationInvalidUrlTest.java]: Done!

File: splunk-quarkus/src/test/java/com/redhat/console/integrations/splunk/SplunkIntegrationNormalTest.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        final String payloadValue = "custom-value";
+        payload.put(payloadKey, payloadValue);
+
+        final JsonObject event = new JsonObject();
+        event.put("metadata", metadata);
+        event.put("payload", payload);
+
+        events.add(event);
+
+        final Exchange invalidSchemeExchange = createExchangeWithBody(cloudEvent.toString());
```
[the remark this answers is not in the document]
Comment (MikelAlejoBR) [reply splunk-quarkus/src/test/java/com/redhat/console/integrations/splunk/SplunkIntegrationNormalTest.java]: Done!

File: splunk-quarkus/src/test/resources/application.properties (hunk 2 above)
[the remark this answers is not in the document]
Comment (gwenneg) [reply splunk-quarkus/src/test/resources/application.properties:11]: Yeah, we'll remove that later if needed.

File: splunk-quarkus/redhat-splunk-quarkus.iml [hunk 6]
```diff
@@ -0,0 +1,354 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (MikelAlejoBR) [reply splunk-quarkus/redhat-splunk-quarkus.iml]: Done! I thought I had removed it :disappointed:

**Review by gwenneg — APPROVED** [review]
LGTM

**Pull request merged**